### PR TITLE
add --no-auto-read flag in params to avoid _read method auto run in constructor

### PR DESCRIPTION
**Pull request by KhalidAriche**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 	<groupId>name.valery1707.kaitai</groupId>
 	<artifactId>kaitai-maven-plugin</artifactId>
-	<version>0.1.7-SNAPSHOT</version>
+	<version>0.1.7</version>
```
Comment (valery1707) [inline pom.xml:9]: Please revert this change - the version will be changed on the release process
```suggestion
	<version>0.1.7-SNAPSHOT</version>
```

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 2]
```diff
@@ -226,6 +227,15 @@ public boolean isNoVersionCheck() {
...
 	}
 
+	/**
+	 * Get no auto read mode.
+	 *
+	 * @return Version no auto read mode
+	 */
+	public boolean isNoAutoRead() {
+		return noAutoRead;
+	}
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:237]: All methods for the single field must be located next to each other

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 3]
```diff
@@ -298,8 +328,8 @@ public Path generate(Logger log) throws KaitaiException {
 
 		log.info("Kaitai: check version");
 		execute(this
-			.process(log)
-			.withArg("--version")
+				.process(log)
+				.withArg("--version")
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:332]: Please revert the indentation changes
```suggestion
			.process(log)
			.withArg("--version")
```

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 4]
```diff
@@ -246,11 +256,31 @@ public KaitaiGenerator noVersionCheck(boolean noVersionCheck) {
...
+	}
+
 	private ProcBuilder process(Logger log) {
 		ProcBuilder builder = new ProcBuilder(getKaitai().normalize().toAbsolutePath().toString())
-			.withErrorStream(new TeeOutputStream(LogWriter.logError(log), streamError))
-			.withOutputStream(new TeeOutputStream(LogWriter.logInfo(log), streamOutput))
-			.withExpectedExitStatuses(0);
+				.withErrorStream(new TeeOutputStream(LogWriter.logError(log), streamError))
+				.withOutputStream(new TeeOutputStream(LogWriter.logInfo(log), streamOutput))
+				.withExpectedExitStatuses(0);
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:283]: Please revert the indentation changes
```suggestion
			.withErrorStream(new TeeOutputStream(LogWriter.logError(log), streamError))
			.withOutputStream(new TeeOutputStream(LogWriter.logInfo(log), streamOutput))
			.withExpectedExitStatuses(0);
```

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 5]
```diff
@@ -276,10 +306,10 @@ private void execute(ProcBuilder builder) throws KaitaiException {
...
 		} catch (StartupException | TimeoutException | ExternalProcessFailureException e) {
 			throw new KaitaiException(
-				"Fail to execute kaitai command: "
-					+ new String(streamError.toByteArray(), UTF_8)
-					+ new String(streamOutput.toByteArray(), UTF_8)
-				, e
+					"Fail to execute kaitai command: "
+							+ new String(streamError.toByteArray(), UTF_8)
+							+ new String(streamOutput.toByteArray(), UTF_8)
+					, e
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:312]: Please revert the indentation changes
```suggestion
				"Fail to execute kaitai command: "
					+ new String(streamError.toByteArray(), UTF_8)
					+ new String(streamOutput.toByteArray(), UTF_8)
				, e
```

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 6]
```diff
@@ -308,9 +338,12 @@ public Path generate(Logger log) throws KaitaiException {
 		}
 
 		ProcBuilder builder = process(log)
-			.withArgs("--target", "java")
-			.withArgs("--outdir", output.toFile().getAbsolutePath())
-			.withArgs("--java-package", getPackageName());
+				.withArgs("--target", "java")
+				.withArgs("--outdir", output.toFile().getAbsolutePath())
+				.withArgs("--java-package", getPackageName());
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:343]: Please revert the indentation changes
```suggestion
			.withArgs("--target", "java")
			.withArgs("--outdir", output.toFile().getAbsolutePath())
			.withArgs("--java-package", getPackageName());
```

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 7]
```diff
@@ -308,9 +338,12 @@ public Path generate(Logger log) throws KaitaiException {
...
+		if (isNoAutoRead()) {
+			builder.withArgs("--no-auto-read");
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:345]: Is this argument is supported by `ksc` version `0.8`?

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 8]
```diff
@@ -27,8 +27,8 @@
  * @see <a href="http://doc.kaitai.io/user_guide.html#_invocation">Kaitai compiler documentation</a>
  */
 @Mojo(
-	name = "generate"
-	, defaultPhase = LifecyclePhase.GENERATE_SOURCES
+		name = "generate"
+		, defaultPhase = LifecyclePhase.GENERATE_SOURCES
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java:31]: Please revert the indentation changes
```suggestion
	name = "generate"
	, defaultPhase = LifecyclePhase.GENERATE_SOURCES
```

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 9]
```diff
@@ -157,6 +157,10 @@
 	@Parameter(property = "kaitai.opaqueTypes")
 	private Boolean opaqueTypes;
 
+
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java:160]: Please add JavaDoc for the new Mojo parameter

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 10]
```diff
@@ -195,18 +200,18 @@ private void executeInt() throws KaitaiException {
 		sourceDirectory = sourceDirectory.toPath().normalize().toFile();
 		if (!sourceDirectory.exists()) {
 			getLog().warn(format(
-				"Skip KaiTai generation: Source directory does not exists: %s"
-				, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
+					"Skip KaiTai generation: Source directory does not exists: %s"
+					, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java:204]: Please revert the indentation changes
```suggestion
				"Skip KaiTai generation: Source directory does not exists: %s"
				, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
```

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 11]
```diff
@@ -195,18 +200,18 @@ private void executeInt() throws KaitaiException {
...
 		if (source.isEmpty()) {
 			getLog().warn(format(
-				"Skip KaiTai generation: Source directory does not contain KaiTai templates (include: %s; exclude: %s): %s"
-				, Arrays.toString(includes)
-				, Arrays.toString(excludes)
-				, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
+					"Skip KaiTai generation: Source directory does not contain KaiTai templates (include: %s; exclude: %s): %s"
+					, Arrays.toString(includes)
+					, Arrays.toString(excludes)
+					, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java:214]: Please revert the indentation changes
```suggestion
				"Skip KaiTai generation: Source directory does not contain KaiTai templates (include: %s; exclude: %s): %s"
				, Arrays.toString(includes)
				, Arrays.toString(excludes)
				, sourceDirectory.toPath().normalize().toFile().getAbsolutePath()
```

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 12]
```diff
@@ -222,15 +227,16 @@ private void executeInt() throws KaitaiException {
...
+				.generator(kaitai, output, packageName)
+				.withSource(source)
+				.overwrite(overwrite)
+				.exactOutput(exactOutput)
+				.executionTimeout(executionTimeout)
+				.fromFileClass(fromFileClass)
+				.opaqueTypes(opaqueTypes)
+				.noVersionCheck(noVersionCheck)
+				.noAutoRead(noAutoRead)
+				.generate(logger);
```
Comment (valery1707) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java:239]: Please revert the indentation changes
```suggestion
			.generator(kaitai, output, packageName)
			.withSource(source)
			.overwrite(overwrite)
			.exactOutput(exactOutput)
			.executionTimeout(executionTimeout)
			.fromFileClass(fromFileClass)
			.opaqueTypes(opaqueTypes)
			.noVersionCheck(noVersionCheck)
			.noAutoRead(noAutoRead)
			.generate(logger);
```

**Review by valery1707 — CHANGES_REQUESTED** [review]

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java (hunk 7 above)
Comment (KhalidAriche) [reply src/main/java/name/valery1707/kaitai/KaitaiGenerator.java:344]: it's supported from 0.9, I need it to avoid creating a private _read method, and calling the read from the constructor.
because I need to override the _read method to handle some exception cases.

**Comment by KhalidAriche** [comment]
@valery1707 Could you please review this PR after my changes ?

**Review by valery1707 — APPROVED** [review]

**Comment by valery1707** [comment]
@rultor merge

**Comment by valery1707** [comment]
@rultor merge
